packages/fxa-auth-server/scripts/recorded-future/check-and-reset.ts

```diff
@@ -77,36 +89,82 @@ Dry run mode is on.  It is the default; use '--dry-run false' when you are ready
 
 Domain: ${searchDomain}
 Filter: first_downloaded_date_gte=${firstDownloadedDateIsoString}
-Limit: 500
+Limit: ${resultsPerPageLimit}
 `);
     return 0;
   }
 
-  const res = await emitStatsdMetrics(
-    async () =>
-      await client.POST('/identity/credentials/search', {
-        body: {
-          domains: [searchDomain],
-          filter: {
-            first_downloaded_gte: firstDownloadedDateIsoString,
-          },
-          limit: 500,
-        },
-      }),
+  const payload = {
+    domains: [searchDomain],
+    filter: {
+      first_downloaded_gte: firstDownloadedDateIsoString,
+    },
+    limit: resultsPerPageLimit,
+  };
+  const _credentialsSearchFn = createCredentialsSearchFn(client);
+  const credentialsSearch = emitStatsdMetrics(
+    _credentialsSearchFn,
     'recorded-future.identity.credentials-search',
     statsd
-  )();
+  );
+  const credentials = await fetchAllCredentialResults(
+    credentialsSearch,
+    payload
+  );
+  statsd.increment(
+    'recorded-future.identity.credentials-search.results.total',
+    credentials.length
+  );
+
+  const emailLoginIdentities = credentials.filter(isLoginAnEmailAddress);
+  statsd.increment(
+    'recorded-future.identity.credentials-search.results.email-logins',
+    emailLoginIdentities.length
+  );
+
+  if (emailLoginIdentities.length === 0) {
+    return 0;
+  }
 
-  if (res.error) {
-    // the Recorded Future Identity OpenAPI definition does not include any
-    // response information aside from the http 200 response
-    throw new Error(`Recorded Future error response: ${(res as any).error}`);
+  const log = logFn({ name: 'recorded-future' });
+  const Token = TokenFn(log, config);
+  const DB = createDB(config, log, Token);
+  const authDb = await DB.connect(config, undefined);
+
+  const getAccountRecord = authDb.accountRecord.bind(authDb);
+  const findAccount = createFindAccountFn(getAccountRecord);
+  const getTotpToken = authDb.totpToken.bind(authDb);
+  const hasTotp2FA = createHasTotp2faFn(getTotpToken);
+
+  const accountEmails: SearchResultIdentity['login'][] = [];
+  for (const identity of emailLoginIdentities) {
+    const acct = await findAccount(identity.login);
+
+    if (acct) {
+      const has2FA = await hasTotp2FA(acct);
+
+      if (!has2FA) {
+        accountEmails.push(identity.login);
+        statsd.increment(
+          'recorded-future.identity.credentials-search.results.account-emails'
+        );
+      } else {
+        statsd.increment(
+          'recorded-future.identity.credentials-search.results.2fa-accounts'
+        );
+      }
+    }
   }
 
-  if (res.data) {
-    console.log(`Found ${res.data.count} results.`);
+  if (accountEmails.length === 0) {
+    console.log('No account emails found.');
+    return 0;
   }
 
+  console.log(
+    `Found ${credentials.length} results, of which ${accountEmails.length} are accounts without 2FA.`
```

Comment:
Do we want to clarify:

`Recorded Future found X results, of which Y are registered accounts and Z are registered accounts without 2FA`

and the "no accounts to reset" case:

`Recorded Future found X results, but none are registered accounts`

Reply:
Oh that output is there right now just so we know the script's working.  Thank you for the suggestions though.